esx_service/vmdk_ops.py

```diff
@@ -402,8 +402,15 @@ def vol_info(vol_meta, vol_size_info, datastore):
 # Return error, or None for OK
 def removeVMDK(vmdk_path):
     logging.info("*** removeVMDK: %s", vmdk_path)
-    cmd = "{0} {1}".format(VMDK_DELETE_CMD, vmdk_path)
 
+    # Check the current volume status
+    attached, uuid, attach_as = getStatusAttached(vmdk_path)
+    if attached:
```

Comment:
Information in KV file could be stale thus preventing user from deleting volume ever. I believe this is a workaround to real bug where Unlink/Delete deletes sidecar (VMFD) but leaves VMDK behind if VMDK is already in use. Does this bug exist with API too?

Also would it be better to query underlying system rather than relying on KV? For e.g. "vmkfstools -D"  tells you whether file is locked and by whom. If it is locked, fail delete?